src/audio/copier/copier.c

```diff
@@ -746,7 +747,8 @@ static int copier_prepare(struct comp_dev *dev)
 		return PPL_STATUS_PATH_STOP;
 
 	for (i = 0; i < cd->endpoint_num; i++) {
-		ret = cd->endpoint[i]->drv->ops.prepare(cd->endpoint[i]);
+		ret = host_prepare_helper(cd->hd);
```

Comment:
@btian1 I think you should do this based on the type of copier or you risk failing prepare for DAI copiers.

Reply:
yes, fully agree, this is just prototype, I am adding the condition on it.